File: core/services/openai_client.py
import os
from openai import OpenAI
from dotenv import load_dotenv, find_dotenv

# print(find_dotenv())

load_dotenv()

client = OpenAI(
    api_key=os.environ.get("OPENAI_API_KEY"),
)

class OpenAIClient:
    """
        Class to provide interface with various interaction with LLM
    """
    def __init__(self, model="gpt-4o"):
        self.model = model

    def generate_text(self, prompt: str) -> str:
        chat_completion = client.chat.completions.create(
            messages=[
                {
                    "role": "user",
                    "content": prompt,
                }
                ],
            model=self.model,
        )
        return chat_completion.choices[0].message.content

    def generate_json(self, prompt: str) -> str:
        chat_completion = client.chat.completions.create(
            messages=[
                {
                    "role": "user",
                    "content": prompt,
                }
                ],
            model=self.model,
            response_format= {
                "type": "json_object"
            }
        )
        return chat_completion.choices[0].message.content

File: src/lib/services/openai-client.ts
import OpenAI from "openai";

/**
 * Class to provide interface with various interactions with LLM
 */
export class OpenAIClient {
  private model: string;
  private client: OpenAI;

  constructor(model: string = "gpt-5-mini-2025-08-07", apiKey?: string) {
    this.model = model;
    this.client = new OpenAI({
      apiKey: apiKey || process.env.OPENAI_API_KEY,
    });
  }

  async generateText(prompt: string): Promise<string> {
    const chatCompletion = await this.client.chat.completions.create({
      messages: [
        {
          role: "user",
          content: prompt,
        },
      ],
      model: this.model,
    });

    return chatCompletion.choices[0].message.content || "";
  }

  async generateJson(prompt: string): Promise<string> {
    const chatCompletion = await this.client.chat.completions.create({
      messages: [
        {
          role: "user",
          content: prompt,
        },
      ],
      model: this.model,
      response_format: {
        type: "json_object",
      },
    });

    return chatCompletion.choices[0].message.content || "{}";
  }
}
